clang/lib/Sema/SemaAttr.cpp

```diff
@@ -269,6 +271,40 @@ void Sema::inferLifetimeBoundAttribute(FunctionDecl *FD) {
   }
 }
 
+static bool IsPointerLikeType(QualType QT) {
```

Comment:
we have a similar function in the CheckExprLifetime.cpp, we can move it to `Sema.h` to make it reusable.